020-Getting-started/100-sveltekit.mdx

```diff
@@ -0,0 +1,629 @@
+```
+
+<Expand title="xata pull main output">
+
+```sh
+$ xata pull main
+Successfully pulled 1 migrations from main branch
+Running codegen...
+Generated Xata code to ./src/xata.ts
+```
+
+</Expand>
+
+This updates the contents of `src/xata.ts` based on the schema defined on the `main` branch of your database. So, if you
+make any further changes to the schema, run `xata pull <branch>` to update the auto-generated code.
+
+## Basic styling and layout
+
+Add the following global styling to the scaffolded `src/app.postcss`:
+
+```css title="app/tailwind.css" {5-26}
+@tailwind base;
+@tailwind components;
+@tailwind utilities;
+
+:root {
+  --foreground-rgb: 0, 0, 0;
+  --background-start-rgb: 214, 219, 220;
+  --background-end-rgb: 255, 255, 255;
+}
+
+@media (prefers-color-scheme: dark) {
+  :root {
+    --foreground-rgb: 255, 255, 255;
+    --background-start-rgb: 0, 0, 0;
+    --background-end-rgb: 0, 0, 0;
+  }
+}
+
+body {
+  color: rgb(var(--foreground-rgb));
+  background: linear-gradient(to bottom, transparent, rgb(var(--background-end-rgb))) rgb(var(--background-start-rgb));
+}
+```
+
+Finally, update `src/routes/+layout.svelte` to add some shared structure across application pages. The code will end up as follows:
+
+```tsx title="src/routes/+layout.svelte" {4-22}
+<script>
+  import '../app.postcss';
+</script>
+<svelte:head>
+    <title>Get started with Xata and SvelteKit</title>
+</svelte:head>
+
+<main class="flex flex-wrap justify-center content-start p-8 lg:p-24 min-h-screen">
+  <div class="z-10 h-50 w-full max-w-5xl items-center justify-between text-xl lg:flex">
+    <p class="fixed left-0 top-0 flex w-full justify-center pb-6 pt-8 lg:static lg:w-auto bg-gradient-to-b from-white via-white via-65% dark:from-black dark:via-black lg:bg-none">
+      <a href="/" data-sveltekit-reload>Get started with Xata and SvelteKit</a>
+    </p>
+    <div class="fixed bottom-0 left-0 flex w-full items-end justify-center bg-gradient-to-t from-white via-white dark:from-black dark:via-black lg:static lg:h-auto lg:w-auto lg:bg-none">
+      <a href="https://xata.io" class="w-20">
+        <img alt="Xata logo" src="https://raw.githubusercontent.com/xataio/examples/main/docs/app_logo.svg" />
+      </a>
+    </div>
+  </div>
+
+  <slot />
+
+</main>
+```
+
+Ensure the `<slot />` component is before the closing `</main>` element.
+
+## Query and list the posts
+
+Now the SvelteKit app is ready for Xata to be integrated into the codebase.
+
+Let's start by making use of
+[SvelteKit's loading data mechanism](https://kit.svelte.dev/docs/load) to retrieve all the blog posts from Xata
+and make them available within the landing page. Create `src/routes/+page.server.ts` with the following contents:
+
+```ts title="src/routes/+page.server.ts"
+import type { PageServerLoad } from './$types';
+
+import { XataClient } from '../xata';
+
+const xata = new XataClient({
+  apiKey: import.meta.env.VITE_XATA_API_KEY,
+  branch: import.meta.env.VITE_XATA_BRANCH
```

Comment:
Small note that not knowing Vite very well I hit an error where it couldn't find the key. This was because I'd skipped the bit earlier in the tutorial (likely because I'm running these tutorials in serial) where you asked to change the `.env` file to include this prefix.

My assumption is people coming to this tutorial will be more familiar and not run into this issue, but I'll bring up we did have a user on Discord this week run into this exact issue, so it might be worth an alert block? I could go either way.

Reply:
There is a paragraph and code snippet dedicated to making this change. So, my opinion is that this is covered in enough detail.

I think your situation of running through a lot of similar steps may have resulted in you skipping this piece.

So, I'd suggest we go with what we have now. But, obviously, keep an eye out for any support/community queries.